polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -3044,30 +3043,37 @@ void MLIRASTConsumer::setMLIRFunctionAttributes(
 
   mlirclang::AttributeList PAL;
   {
+    const clang::CodeGen::CGFunctionInfo &FI = GetOrCreateCGFunctionInfo(&FD);
     const auto *FPT = FD.getType()->getAs<FunctionProtoType>();
     clang::CodeGen::CGCalleeInfo CalleeInfo(FPT);
 
+    unsigned CallingConv;
     getTypes().constructAttributeList(function.getName(), FI, CalleeInfo, PAL,
-                                      /*AttrOnCallSite=*/false,
+                                      CallingConv,
+                                      /*AttrOnCallSite*/ false,
                                       /*IsThunk*/ false);
 
     // Set additional function attributes that are not derivable from the
     // function declaration.
-    mlirclang::AttrBuilder attrBuilder(*ctx);
+    mlirclang::AttrBuilder attrBuilder(*Ctx);
     {
-      LLVM::Linkage lnk = getMLIRLinkage(getLLVMLinkageType(FD, ShouldEmit));
+      attrBuilder.addAttribute(
```

Comment:
Used to be set in `constructAttributeList`, that function should only produce information (not change the IR).